core/variant/variant_call.cpp

```diff
@@ -806,6 +806,34 @@ struct _VariantCall {
 		return dest;
 	}
 
+	static PackedVector2Array func_PackedByteArray_decode_vector2_array(PackedByteArray *p_instance) {
+		uint64_t size = p_instance->size();
+		PackedVector2Array dest;
+		if (size == 0) {
+			return dest;
+		}
+		ERR_FAIL_COND_V_MSG(size % sizeof(Vector2), dest, "PackedByteArray size must be a multiple of 8 (size of Vector2) to convert to PackedFloat32Array.");
```

Comment:
This code looks like it should handle `real_t` set to `double`, except that the text in this message becomes wrong.

Reply:
Right, yes. The messages are hardcoded in the other to_X methods, and I thought it was a bit silly to invoke `snprintf` for this, especially as it can be resolved at compile time… I guess I could do some kind of voodoo compile-time string concatenation? Don't suppose there's already a macro for this somewhere in the codebase…?

I suppose this does mean that these methods can't necessarily be used to store data that will be used across builds/platforms, which we should maybe note in the documentation.

Reply:
How about just this?

```cpp
"PackedByteArray size must be a multiple of Vector2 size to convert to PackedVector2Array."
```

Reply:
oh… yeah, that seems kind of obvious in retrospect haha